autoload/easy_align.vim

```diff
@@ -89,9 +89,33 @@ function! s:floor2(v)
   return a:v % 2 == 0 ? a:v : a:v - 1
 endfunction
 
+function! s:get_highlight_group_name(line, col)
+  let l:hl = synIDattr(synID(a:line, a:col, 0), 'name')
+
+  if l:hl == '' && has('nvim')
+    " try getting the highlighting group using neovim treestiiter approach.
+	" The required functionality appeared in neovim nightly around 18 Dec, 2022
+	" https://www.reddit.com/r/neovim/comments/zou5ig/inspect_is_now_available_on_nightly/
+	let insp = luaeval( 'vim.inspect_pos and vim.inspect_pos( nil, ' .. (a:line-1) .. ', ' .. (a:col-1) .. ' ) or { treesitter = {} }' )
+    if ! empty( insp.treesitter )
+      let l:hl = insp.treesitter[0].hl_group_link
+    endif
+  endif
+
+  " and, finally
+  return l:hl
+endfunction
+
+function! easy_align#get_highlight_group_name(line = line('.'), col = col('.'))
```

Comment:
I don't think this syntax works in older versions of Vim. vim-easy-align was created a long time ago, and hasn't gone through a lot of updates lately, users would expect it to work on relatively older versions of Vim.

https://github.com/vim/vim/commit/42ae78cfff171fbd7412306083fe200245d7a7a6

Please change it to `...`, or remove the optional arguments and make the function only report the highlighting group of the current position.

https://vi.stackexchange.com/a/11548/661

Reply:
Updated, please review